### ISSUE TITLE
Bug in fetch_and_shift_spectra()

### ISSUE THREAD

**Issue by kareemelbadry** [astroML/astroML#60]
I'm pretty sure there's a bug here: http://www.astroml.org/examples/datasets/compute_sdss_pca.html#example-datasets-compute-sdss-pca

This condition:

```
    if np.all(spec_rebin.spectrum == 0):
        num_skipped += 1
        print "%i, %i, %i is all zero" % (plate[i], mjd[i], fiber[i])
        continue
```

needs an "i += 1" added before the continue statement. 

As it is now, if a bad spectrum (or one outside the given wavelength range) is found, no progress is made, because the function keeps trying to get the same spectrum over and over, hitting the same condition each time without incrementing i.

**Comment by jakevdp** [comment]
Thanks – you're correct. This would result in an infinite loop if the condition were ever met – that's not happened when I've run the code, so I guess I got lucky! Would you like to submit a PR with the fix?

**Comment by bsipocz** [comment]
addressed by #85
